core/src/main/java/nu/yona/server/analysis/entities/GoalConflictMessage.java

```diff
@@ -125,22 +119,17 @@ public static GoalConflictMessage createInstanceFromBuddy(UUID relatedUserAnonym
 	{
 		if (origin == null)
 		{
-			throw GoalServiceException.originCannotBeNull();
+			throw new IllegalArgumentException("origin cannot be null");
```

Comment:
There are still 12 instances of `IllegalArgumentException` in the source code. I propose not to change all of those with this design change because it clutters up the Pull Request. It's OK to remove them, so please add a separate task.